builtin/mainmenu/modmgr.lua

```diff
@@ -286,21 +286,36 @@ end
 function modmgr.get_dependencies(modfolder)
 	local toadd = ""
 	if modfolder ~= nil then
-		local filename = modfolder ..
-					DIR_DELIM .. "depends.txt"
-
-		local dependencyfile = io.open(filename,"r")
-
-		if dependencyfile then
-			local dependency = dependencyfile:read("*l")
-			while dependency do
+		local mod_conf = Settings(modfolder .. DIR_DELIM .. "mod.conf")
+		local dependencies = nil
+		if mod_conf then
+			dependencies = mod_conf:get("depends")
+		end
+		if dependencies then
+			local list = dependencies:split()
+			for i, element in ipairs(list) do
 				if toadd ~= "" then
 					toadd = toadd .. ","
 				end
-				toadd = toadd .. dependency
-				dependency = dependencyfile:read()
+				toadd = toadd .. element:trim()
+			end
+		else
+			local filename = modfolder ..
+						DIR_DELIM .. "depends.txt"
+
+			local dependencyfile = io.open(filename,"r")
+
+			if dependencyfile then
+				local dependency = dependencyfile:read("*l")
+				while dependency do
```

Comment:
While you're changing things around here, can you change this to use `:lines()`?